### PR TITLE
revert (core): ChrRace.dbc full implementation

### PR DESCRIPTION
**Pull request by acidmanifesto**
we revert this due to several issues arrising. Although the dbc reading is done in full and correctly. Azerothcore relied on the original handling (althought not propper) for so long that  there is reliance on it
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by mpfans** [comment]
> Please merge this commit immediately as it disrupts many functions, such as ineffective Druid transformation, inability to add offline characters as friends, and inability to mail items to offline players

I have tested and accepted this PR, everything is back to work

**Comment by acidmanifesto** [comment]
Current failure is with eluna.
Name -> name

File: data/sql/updates/pending_db_world/remove_charrace_dbc.sql [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+-- Dumping data for table acore_world.chrclasses_dbc: ~0 rows (approximately)
+DELETE FROM `chrclasses_dbc`;
+
+/*!40103 SET TIME_ZONE=IFNULL(@OLD_TIME_ZONE, 'system') */;
+/*!40101 SET SQL_MODE=IFNULL(@OLD_SQL_MODE, '') */;
+/*!40014 SET FOREIGN_KEY_CHECKS=IFNULL(@OLD_FOREIGN_KEY_CHECKS, 1) */;
+/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
+/*!40111 SET SQL_NOTES=IFNULL(@OLD_SQL_NOTES, 1) */;
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/remove_charrace_dbc.sql:90]:
```
DROP TABLE IF EXISTS `chrraces_dbc`;
CREATE TABLE `chrraces_dbc` (
  `ID` int(11) NOT NULL DEFAULT 0,
  `Flags` int(11) NOT NULL DEFAULT 0,
  `FactionID` int(11) NOT NULL DEFAULT 0,
  `ExplorationSoundID` int(11) NOT NULL DEFAULT 0,
  `MaleDisplayId` int(11) NOT NULL DEFAULT 0,
  `FemaleDisplayId` int(11) NOT NULL DEFAULT 0,
  `ClientPrefix` varchar(100) DEFAULT NULL,
  `BaseLanguage` int(11) NOT NULL DEFAULT 0,
  `CreatureType` int(11) NOT NULL DEFAULT 0,
  `ResSicknessSpellID` int(11) NOT NULL DEFAULT 0,
  `SplashSoundID` int(11) NOT NULL DEFAULT 0,
  `ClientFilestring` varchar(100) DEFAULT NULL,
  `CinematicSequenceID` int(11) NOT NULL DEFAULT 0,
  `Alliance` int(11) NOT NULL DEFAULT 0,
  `Name_Lang_enUS` varchar(100) DEFAULT NULL,
  `Name_Lang_enGB` varchar(100) DEFAULT NULL,
  `Name_Lang_koKR` varchar(100) DEFAULT NULL,
  `Name_Lang_frFR` varchar(100) DEFAULT NULL,
  `Name_Lang_deDE` varchar(100) DEFAULT NULL,
  `Name_Lang_enCN` varchar(100) DEFAULT NULL,
  `Name_Lang_zhCN` varchar(100) DEFAULT NULL,
  `Name_Lang_enTW` varchar(100) DEFAULT NULL,
  `Name_Lang_zhTW` varchar(100) DEFAULT NULL,
  `Name_Lang_esES` varchar(100) DEFAULT NULL,
  `Name_Lang_esMX` varchar(100) DEFAULT NULL,
  `Name_Lang_ruRU` varchar(100) DEFAULT NULL,
  `Name_Lang_ptPT` varchar(100) DEFAULT NULL,
  `Name_Lang_ptBR` varchar(100) DEFAULT NULL,
  `Name_Lang_itIT` varchar(100) DEFAULT NULL,
  `Name_Lang_Unk` varchar(100) DEFAULT NULL,
  `Name_Lang_Mask` int(10) unsigned NOT NULL DEFAULT 0,
  `Name_Female_Lang_enUS` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_enGB` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_koKR` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_frFR` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_deDE` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_enCN` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_zhCN` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_enTW` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_zhTW` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_esES` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_esMX` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_ruRU` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_ptPT` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_ptBR` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_itIT` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_Unk` varchar(100) DEFAULT NULL,
  `Name_Female_Lang_Mask` int(10) unsigned NOT NULL DEFAULT 0,
  `Name_Male_Lang_enUS` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_enGB` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_koKR` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_frFR` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_deDE` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_enCN` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_zhCN` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_enTW` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_zhTW` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_esES` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_esMX` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_ruRU` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_ptPT` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_ptBR` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_itIT` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_Unk` varchar(100) DEFAULT NULL,
  `Name_Male_Lang_Mask` int(10) unsigned NOT NULL DEFAULT 0,
  `FacialHairCustomization_1` varchar(100) DEFAULT NULL,
  `FacialHairCustomization_2` varchar(100) DEFAULT NULL,
  `HairCustomization` varchar(100) DEFAULT NULL,
  `Required_Expansion` int(11) NOT NULL DEFAULT 0,
  PRIMARY KEY (`ID`)
) ENGINE=MyISAM DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;
```

Comment (pklloveyou) [reply data/sql/updates/pending_db_world/remove_charrace_dbc.sql:90]: Hello, you seem to have made a mistake in the table name

**Comment by Gultask** [comment]
Don't you need to re-add the previous table?
https://github.com/azerothcore/azerothcore-wotlk/blob/master/data/sql/base/db_world/chrraces_dbc.sql

File: data/sql/updates/pending_db_world/remove_charrace_dbc.sql (hunk 1 above)
Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/remove_charrace_dbc.sql:90]: What mistake? Its the original sql from the squash 2 days ago.

Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/remove_charrace_dbc.sql:90]: Ah wait i see. Wrong sql. I will correct

Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/remove_charrace_dbc.sql]: Done. Thank you 👍

**Review by Nyeriah — APPROVED** [review]

**Comment by zjgzh001** [comment]
大神们好！最近编译新的AzerothCore，出现以下错误：
Wrong format occurred (invalid format string) at 'E:\azbotcoreserever\azerothcore-wotlk\src\server\game\Globals\ObjectMgr.cpp:2888'

望大神们指定！非常感谢
大神们好！最近编译新的AzerothCore，出现以下错误：
Wrong format occurred (invalid format string) at 'E:\azbotcoreserever\azerothcore-wotlk\src\server\game\Globals\ObjectMgr.cpp:2888'

望大神们指定！非常感谢
Hello, great gods! Recently, a new AzerothCore was compiled with the following error:

Wrong format occurred (invalid format string) at 'E:\azbotcoreserever\azerothcore-wotlk\src\server\game\Globals\ObjectMgr.cpp:2888'

May the great gods designate! Thank you.

**Comment by zjgzh001** [comment]
Hello, great gods! Recently, a new AzerothCore was compiled with the following error:
Wrong format occurred (invalid format string) at 'E:\azbotcoreserever\azerothcore-wotlk\src\server\game\Globals\ObjectMgr.cpp:2888'
May the great gods designate! Thank you.

**Comment by zjgzh001** [comment]
The newly downloaded azbotcore compiled without errors. When starting Worldserver after installation, the following error occurred:

Wrong format occurred (invalid format string) at 'E:\azbotcoreserever\azerothcore-wotlk\src\server\game\Globals\ObjectMgr.cpp:2888'

Can the bosses help repair it?
QQ截图20230511151756

**Comment by Nyeriah** [comment]
Please open a new issue if you are experiencing issues. This has nothing to do with this PR

**Pull request merged**